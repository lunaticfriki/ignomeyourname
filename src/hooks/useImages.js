import { useState } from 'react'

const useImages = () => {
  const [imagesDisplayed, setImagesDisplayed] = useState(0)
  const [moreImages, setMoreImages] = useState(24)

  const previousImages = () => {
    setImagesDisplayed(imagesDisplayed - 24)
    setMoreImages(moreImages - 24)
  }
  const nextImages = () => {
    setImagesDisplayed(imagesDisplayed + 24)
    setMoreImages(moreImages + 24)
  }
  return { imagesDisplayed, moreImages, previousImages, nextImages }
}

export default useImages
