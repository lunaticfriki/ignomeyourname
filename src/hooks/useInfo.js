import React, { useState, useRef } from 'react'
import PropTypes from 'prop-types'

import Info from '../components/info'
import ModalInfo from '../components/modalInfo'
import useOnClick from './useOnClick'

const useInfo = (gnomes, gnome) => {
  const [showModal, setShowModal] = useState(false)

  const viewInfo = () => {
    setShowModal(!showModal)
  }

  const ref = useRef()
  useOnClick(ref, () => setShowModal(false))

  const Modal = React.forwardRef((props, ref) => {
    return (
      <ModalInfo ref={ref}>
        <header>
          <button onClick={() => setShowModal(false)}>
            <i className="fas fa-times"></i>
          </button>
        </header>
        <section>
          <Info gnomes={gnomes} gnome={gnome.name} />
        </section>
      </ModalInfo>
    )
  })
  return { viewInfo, Modal, showModal }
}

export default useInfo

useInfo.propTypes = {
  gnomes: PropTypes.array,
  gnome: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  setGnome: PropTypes.func,
}

useInfo.defaultProps = {
  gnomes: [],
  gnome: '',
  setGnome: () => {},
}
