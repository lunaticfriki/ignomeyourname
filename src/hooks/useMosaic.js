import { useState } from 'react'

const useMosaic = () => {
  const [mosaic, setMosaic] = useState(false)

  const showMosaic = () => {
    setMosaic(!mosaic)
  }
  return { showMosaic, mosaic }
}

export default useMosaic
