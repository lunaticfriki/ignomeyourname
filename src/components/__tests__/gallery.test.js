import React from 'react'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect'
import { renderHook, act } from '@testing-library/react-hooks'

import Gallery from '../gallery'
import useImages from '../../hooks/useImages'
import useMosaic from '../../hooks/useMosaic'
import useInfo from '../../hooks/useInfo'

describe('Gallery component suite', () => {
  const dummyProps = {
    gnomes: [],
    gnome: {},
    setGnome: jest.fn(),
  }

  afterEach(() => {
    jest.clearAllMocks()
  })

  test('Gallery snapshot', () => {
    const component = renderer.create(<Gallery {...dummyProps} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('Should trigger a function to show next images', () => {
    const { result } = renderHook(() => useImages())

    expect(result.current.imagesDisplayed).toBe(0)
    expect(result.current.moreImages).toBe(24)
    act(() => result.current.nextImages())
    expect(result.current.imagesDisplayed).toBe(24)
    expect(result.current.moreImages).toBe(48)
  })
  test('Should trigger a function to show previous images', () => {
    const { result } = renderHook(() => useImages())

    expect(result.current.imagesDisplayed).toBe(0)
    expect(result.current.moreImages).toBe(24)
    act(() => result.current.previousImages())
    expect(result.current.imagesDisplayed).toBe(-24)
    expect(result.current.moreImages).toBe(0)
  })
  test('Should trigger a function to display mosaic view', () => {
    const { result } = renderHook(() => useMosaic())
    expect(result.current.mosaic).toBe(false)
    act(() => result.current.showMosaic())
    expect(result.current.mosaic).toBe(true)
  })
  test('Should trigger a function to display modal info', () => {
    const { result } = renderHook(() => useInfo())
    expect(result.current.showModal).toBe(false)
    act(() => result.current.viewInfo())
    expect(result.current.showModal).toBe(true)
  })
})
