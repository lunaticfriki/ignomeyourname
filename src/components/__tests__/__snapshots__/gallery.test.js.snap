// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery component suite Gallery snapshot 1`] = `
<div
  className="sc-AxjAm gxJPRT"
>
  <div
    className="c-gallery-menu"
  >
    <button
      onClick={[Function]}
    >
      Show next
    </button>
    <button
      className="c-button-mosaic"
      data-testid="mosaic"
      id="mosaic"
      onClick={[Function]}
    >
      <i
        className="fas fa-th"
      />
    </button>
  </div>
  <section
    className="c-display"
  />
  <div
    className="c-gallery-menu"
  >
    <div>
      <button
        onClick={[Function]}
      >
        Show more
      </button>
    </div>
  </div>
</div>
`;
